engine/engine.go

```diff
@@ -191,3 +229,28 @@ func (e *Engine) runTask(ctx context.Context, task tasks.Task) {
 		tlog.Errorw("error updating final status", "err", err)
 	}
 }
+
+func getTaskSchedule(task tasks.Task) (string, time.Duration) {
+	var schedule, limit string
+	var duration time.Duration
+
+	if task.RetrievalTask.Exists() {
+		schedule = task.RetrievalTask.Must().Schedule.Must().String()
+		limit = task.StorageTask.Must().ScheduleLimit.Must().String()
```

Comment:
did you mean RetrievalTask here? I think you would avoid mistakes like these with...

```
if t := task.RetrievalTask; t.Exists {
    schedule = t.Must()...
```

Reply:
Much nicer.  Also, obviously, I did not test the retrieval task with a limit.